src/history/hs_rec.c

```diff
@@ -160,29 +158,25 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
     }
 
     /*
-     * If we're inserting a non-zero timestamp, look ahead for any higher timestamps. If we find
-     * updates, we should remove them and reinsert them at the current timestamp.
+     * Look ahead for any higher timestamps. If we find updates, we should remove them and reinsert
+     * them at the current timestamp. If there were no keys equal to or less than our target key, we
+     * would have received WT_NOT_FOUND. In that case we need to search again with a higher
+     * timestamp.
      */
-    if (tw->start_ts != WT_TS_NONE) {
-        /*
-         * If there were no keys equal to or less than our target key, we would have received
-         * WT_NOTFOUND. In that case we need to search again with a higher timestamp as the cursor
-         * would not be positioned correctly.
-         */
-        if (ret == 0)
-            WT_ERR_NOTFOUND_OK(cursor->next(cursor), true);
-        else {
-            F_SET(cursor, WT_CURSTD_HS_READ_ALL);
+    if (ret == 0)
```

Comment:
Just noticed that we are doing another search_near() in `__hs_delete_reinsert_from_pos`. Can we somehow simplify the logic and here probably get rid of next() or search_near() call here?

Reply:
Actually scratch that - I mixed up the function with __wt_hs_delete_reinsert_key_from_ts()